### ISSUE TITLE
Error reading argo file when using `read.oce()`

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1118]
I don't know how general this example is, but I just tried loading a particular Argo file using `read.oce()` and got the following error:
```r
library(oce)
dd <- read.oce('nodc_6901486_prof.nc')
Error in R_nc4_get_vara_text: NetCDF: Index exceeds dimension bound
Var: history_institution  Ndims: 3   Start: 0,0,0 Count: 0,299,4
 ```
with traceback
```r
5: stop("C function R_nc4_get_var_text returned error")
4: ncvar_get_inner(ncid2use, varid2use, nc$var[[li]]$missval, addOffset, 
       scaleFact, start = start, count = count, verbose = verbose, 
       signedbyte = signedbyte, collapse_degen = collapse_degen)
3: ncdf4::ncvar_get(f, name) at oce.R#1693
2: read.netcdf(file, ...) at oce.R#1649
1: read.oce("nodc_6901486_prof.nc")
```
This makes it look like a problem with the `ncdf4` package, except that if instead of using `read.oce()` I use `read.argo()`, it works fine:
```r
dd <- read.argo('nodc_6901486_prof.nc')
```

The file is a bit big (~60 MB) but can be found here (in the shared Dropbox):
https://www.dropbox.com/s/b44np1nf3ontb5c/nodc_6901486_prof.nc?dl=0

**Comment by dankelley** [comment]
I'll look this aft. PS. netcdf really is cool. This dataset only takes 0.6s to read, on my home computer (work would be slower).

**Comment by dankelley** [comment]
**DIARY-TYPE ENTRY -- I will keep editing this as I work**

Oh, I see. It is trying to read a general netcdf file. It doesn't know that it is an argo file. That's why `read.oce()` is doing differently than `read.argo()` is. The decision is made near line 1648 of `R/oce.R`. Maybe at that spot in the code we should be looking at the file contents to try to see just what kind of netcdf file it is.

Hm. I put in a browser command and I see that the netcdf code is simply unable to read the variable called `history_institution`. I think thats a ncdf4 error, not an oce error.
```R
> ncdf4::ncvar_get(f, "history_institution")
Error in R_nc4_get_vara_text: NetCDF: Index exceeds dimension bound
Var: history_institution  Ndims: 3   Start: 0,0,0 Count: 0,299,4
Error in ncvar_get_inner(ncid2use, varid2use, nc$var[[li]]$missval, addOffset,  : 
  C function R_nc4_get_var_text returned error
```

Fine, ok, but what to *do*? Wrapping the `ncvar_get` call in a `try` does not help because the error is down at the C level.  I suppose we could just *skip* that variable (which is not being used by oce anyhow). Hm.

**Comment by dankelley** [comment]
OK, I have this working now if I do as follows (near line 1694 of `R/oce.R`).
```R
 if (name == "history_institution" || name == "history_step" || name == "history_software"
            || name == "history_software_release" || name == "history_reference" || name == "history_date"
            || name == "history_action" || name == "history_parameter" || name == "history_start_pres"
            || name == "history_stop_pres" || name == "history_previous_value"
            || name == "history_qctest")
            next
```
In other words, I found that R cannot read any of these named variables. Sheesh. I do not know why they are a problem. I used the unix utility `ncdump` and found the following, for example.
```R
        char history_institution(n_history, n_prof, string4) ;
                history_institution:long_name = "Institution which performed action" ;
                history_institution:conventions = "Argo reference table 4" ;
                history_institution:_FillValue = " " ;
```

**Comment by dankelley** [comment]
CLARK -- some questions for you here.

*Status* I can read the file (the code is not yet committed though) and that's good, in the sense that I don't want users to be seeing C-level errors. However, there are two issues, on which I'd appreciate Clark's input.

1.  The code I've developed is a bit scary to me. I don't like skipping a list of named entries. I might prefer to just skip *any* entry with a name beginning `history_`. Clark, what do you think about that?

2. I think we don't want to spend a lot of effort on point 1, because the object that gets read will not be in `argo` class, and `as.argo()` cannot convert it, either.  (It's just a bit weird thing.) What I propose, instead, is that `oceMagic()` should peek inside netcdf files, and if they seem to be argo files, it should call `read.argo()`. (Recall, there are other files ... lots of people save lots of data in netcdf, so we certainly don't want all netcdf files to go to `read.argo`.)  My question for Clark: do you know of any string that we can rely upon, to tell us that a netcdf file contains argo data? There are a bunch of things in this *particular* file that suggest it's an argo (see below) but Clark do you know of any we can rely on?  I think you've maybe looked at the data docs more recently than I have done.
```
$ ncdump -h nodc_6901486_prof.nc | grep -i argo
		data_type:conventions = "Argo reference table 1" ;
		station_parameters:conventions = "Argo reference table 3" ;
		data_centre:conventions = "Argo reference table 4" ;
		data_state_indicator:conventions = "Argo reference table 6" ;
		platform_type:conventions = "Argo reference table 23" ;
		wmo_inst_type:conventions = "Argo reference table 8" ;
		juld_qc:conventions = "Argo reference table 2" ;
		position_qc:conventions = "Argo reference table 2" ;
		profile_pres_qc:conventions = "Argo reference table 2a" ;
		profile_temp_qc:conventions = "Argo reference table 2a" ;
		profile_psal_qc:conventions = "Argo reference table 2a" ;
		vertical_sampling_scheme:conventions = "Argo reference table 16" ;
		pres_qc:conventions = "Argo reference table 2" ;
		pres_adjusted_qc:conventions = "Argo reference table 2" ;
		temp_qc:conventions = "Argo reference table 2" ;
		temp_adjusted_qc:conventions = "Argo reference table 2" ;
		psal_qc:conventions = "Argo reference table 2" ;
		psal_adjusted_qc:conventions = "Argo reference table 2" ;
		parameter:conventions = "Argo reference table 3" ;
		history_institution:conventions = "Argo reference table 4" ;
		history_step:conventions = "Argo reference table 12" ;
		history_action:conventions = "Argo reference table 7" ;
		history_parameter:conventions = "Argo reference table 3" ;
		:title = "Argo float vertical profile" ;
		:source = "Argo float" ;
		:references = "http://www.nodc.noaa.gov/argo/" ;
		:Conventions = "GADR-3.0 Argo-3.0 CF-1.6" ;
		:summary = "The U.S. National Oceanographic Data Center (NODC) operates the Argo Global Data Repository (GADR). For information about organizations contributing data to GADR, see http://www.nodc.noaa.gov/argo/" ;
		:file_source = "The Argo Global Data Assembly Center FTP server at ftp://ftp.ifremer.fr/ifremer/argo" ;
```

**Comment by dankelley** [comment]
Oh, crap. I've seen this before, and forgot. In `R/oce.R` we check to see if there is a variable called `DATA_TYPE`, and that is how we know it's `argo`.  However, in this file, that variable is named `data_type`.  I'm going to change the test to also look for the lower-case version.

**Comment by dankelley** [comment]
I think this is fixed in `develop` commit 83d676877fb2a388cb67dd2eeae3eee035e3ca0e

1. it works on this file (well, it reads it -- I didn't look at the data)
2. it still reads our existing `argo` dataset

Q: seem OK, Clark?

**Comment by richardsc** [comment]
Seems great to me! Thanks buddy :smile:
